From dc0d80b90df10ce459c278799b83db0f7e334542 Mon Sep 17 00:00:00 2001
From: nift4 <nift4@protonmail.com>
Date: Fri, 18 Aug 2023 00:34:54 +0200
Subject: [PATCH 4/5] Enable NLP when emergency call is active

The update function is called by
8c3cd9f68cabba64f91f962264029c66958b2e6b, so this is applied instantly
---
 smali/com/mediatek/location/lppe/main/LPPeService.smali | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/smali/com/mediatek/location/lppe/main/LPPeService.smali b/smali/com/mediatek/location/lppe/main/LPPeService.smali
index 9c591c5..9363e0f 100644
--- a/smali/com/mediatek/location/lppe/main/LPPeService.smali
+++ b/smali/com/mediatek/location/lppe/main/LPPeService.smali
@@ -3021,10 +3021,17 @@
     .locals 3
 
     .line 1622
+    invoke-direct {p0}, Lcom/mediatek/location/lppe/main/LPPeService;->isInEmergency()Z
+
+    move-result v0
+
+    if-nez v0, :cond_1
+
     invoke-direct {p0}, Lcom/mediatek/location/lppe/main/LPPeService;->isAssistedGpsEnabled()Z
 
     move-result v0
 
+    :cond_1
     .line 1624
     new-instance v1, Ljava/lang/StringBuilder;
 
-- 
2.56.0

